Remove ripplerest (and related)

mDuo13 opened the pull request:
This massive overhaul of dev portal content removes references to Ripple-REST (which has been discontinued). In many cases, such references are replaced with references to its successor, RippleAPI. In some cases, there was too much and the references to REST were simply removed with no substitute. (I may later try to revisit those sections to add RippleAPI versions, but that's a low priority.)

This entirely removes the Ripple-REST article and the corresponding Ripple-REST API Tool.

Along the way, I also removed some other deprecated things, including the demurrage article (demurrage code has been removed) and a number of HTML files that were hanging around despite being removed (e.g. gateway services).

mDuo13 committed:
start removing Ripple-REST

mDuo13 committed:
gateway guide - more examples in rippled format

mDuo13 committed:
parse_pages - fix typo in watch syntax

mDuo13 committed:
gateway guide - removed the rest of the REST examples

mDuo13 committed:
gateway guide - clarification on partial payments in RippleAPI

mDuo13 commented on content/gateway_guide.md:
````diff
@@ -395,110 +400,123 @@ _(**Reminder:** Don't send your secret to a server you do not control.)_
 ```
-_(**Reminder:** Don't send your secret to a server you do not control.)_
-
-Response:
````
Known issue: The updated document doesn't have an example response. That's because I gave up after several failed test cases. (part of that was because I discovered RIPD-1073)

mDuo13 commented on content/gateway_guide.md:
````diff
@@ -506,46 +524,37 @@ You may also enable `RequireAuth` for your cold wallet in order to use the [Auth
 }
 ```
 
-A successful response shows `"state": "validated"` when the change has been accepted into a validated Ripple ledger.
````
Known issue: This example is also missing a response, because I didn't have a successful test case. I can probably add one here later.

alandotcom commented:
@mDuo13 should I be able to checkout this PR and open `index.html` locally to see the rendered changes?

mDuo13 commented:
@lumberj Yes, I'm fairly confident that will work.

alandotcom commented:
I see Ripple-REST on the title page

![screen shot 2016-01-26 at 4 22 56 pm](https://cloud.githubusercontent.com/assets/1812196/12599714/247031d6-c449-11e5-8d03-ca232cd5b962.png)

mDuo13 committed:
Redo index page to resemble build index

mDuo13 commented:
@lumberj Good catch. The "index.html" page is not really used anymore, so I had forgotten to update it. I added another commit that changes the template for that page so it generates an index that's closer to what we have on ripple.com/build/ now.

ryangyoung commented on content/gateway_guide.md:
```diff
@@ -506,46 +524,37 @@ You may also enable `RequireAuth` for your cold wallet in order to use the [Auth
 
 If ACME decides to use Authorized Accounts, ACME creates an interface for users to get their Ripple trust lines authorized by ACME's cold account. After Alice has extended a trust line to ACME from her Ripple account, she goes through the interface on ACME's website to require ACME authorize her trust line. ACME confirms that it has validated Alice's identity information, and then sends a TrustSe ...
```
This seems out of place here. Wouldn't it make more sense to move this paragraph to the beginning of the RequireAuth section? Then the structure of the "With Cold Wallets" and "With Hot Wallets" sections are parallel.

mDuo13 replied:
There's probably a good way to reorganize this, but it's a little tricky because we are using RequireAuth for something other than its primary purpose. (We recommend that you enable RequireAuth on hot wallets even if you don't want to use the Authorized Accounts feature, because that way your hot wallets can never accidentally issue currency unless they also accidentally authorize trust lines first.)

ryangyoung replied:
It's just a little disconcerting to see a paragraph about how ACME should generally use RequireAuth in a subsection of called "With Cold Wallets". It would make more sense to me to see it in the general introduction to the feature, whether or not that feature is being used as intended.

mDuo13 replied:
Agreed. This is not really related to the use of Ripple-REST, but I can make the change anyway.

mDuo13 replied:
I modified the organization of this section. I think it's clearer in the updated version.

ryangyoung commented on content/gateway_guide.md:
```diff
@@ -553,103 +562,158 @@ In order to robustly monitor incoming payments, gateways should do the following
 * Keep a record of the most-recently-processed transaction and ledger. That way, if you temporarily lose connectivity, you know how far to go back.
 * Check the result code of every incoming payment. Some payments go into the ledger to charge an anti-spam fee, even though they failed. Only transactions with the result code `tesSUCCESS` can change non-XRP balances. Only transactions from a validated ledger are final.
-* [Look out for Partial Payments](https://ripple.com/files/GB-2014-06.pdf "Partial Payment Flag Gateway Bulletin"). If an incoming transaction has a `destination_balance_changes` field (Ripple-REST) or a `meta.delivered_amount` field (WebSocket/JSON-RPC), then use that to see how much money *actually* got delivered to the destination account. Payments with the partial-payment flag enabled are con ...
+* [Look out for Partial Payments](https://ripple.com/files/GB-2014-06.pdf "Partial Payment Flag Gateway Bulletin"). Payments with the partial-payment flag enabled can be considered "successful" if any non-zero amount is delivered, even miniscule amounts.
```
Let's send them to the KC instead of to the PDF: https://ripple.com/knowledge_center/partial-payment-flag/

mDuo13 replied:
Will do.

ryangyoung commented on content/gateway_guide.md:
```diff
@@ -658,64 +722,15 @@ When your hot or cold wallet receives a payment whose purpose is unclear, we rec
 
 The first requirement to bouncing payments is [robustly monitoring for incoming payments](#robustly-monitoring-for-payments). You do not want to accidentally refund a user for more than they sent you! (This is particularly important if your bounce process is automated.) The [Partial Payment Flag Gateway Bulletin](https://ripple.com/files/GB-2014-06.pdf) explains how to avoid a common problem.
```
Let's use the link to the KC for the partial payment GB: https://ripple.com/knowledge_center/partial-payment-flag/

mDuo13 replied:
Roger.

ryangyoung commented on content/reliable_tx.md:
```diff
@@ -17,7 +17,7 @@ These types of errors can potentially lead to serious problems.  For example, an
 ## Background
 
-The Ripple protocol provides a ledger shared across all nodes in the network.  Through a process of consensus and validation, the network agrees on order in which transactions are applied to (or omitted from) the ledger.  <!--See Ripple Ledger Consensus and Validation[b] for an overview of this process.-->
+The Ripple protocol provides a ledger shared across all nodes in the network.  Through a process of consensus and validation, the network agrees on order in which transactions are applied to (or omitted from) the ledger.
```
Maybe add a link to the Consensus article in the KC? https://ripple.com/knowledge_center/the-ripple-ledger-consensus-process/

mDuo13 replied:
Added.

ryangyoung commented on content/reliable_tx.md:
```diff
@@ -28,33 +28,37 @@ Applications face additional challenges, in the event of power or network loss,
-While applying transactions, Ripple network nodes work with the *last validated ledger*, a snapshot of the ledger state based on transactions the entire network has validated.  The process of consensus and validation apply a set of new transactions to the last validated ledger, resulting in a new validated ledger.  This new validated ledger instance and the ones that preceded it comprise the ledg ...
+While applying transactions, `rippled` servers use the *last validated ledger*, a snapshot of the ledger state based on transactions the entire network has validated.  The process of consensus and validation apply a set of new transactions to the last validated ledger in canonical order, resulting in a new validated ledger.  This new validated ledger instance and the ones that preceded it compris ...
 
+Each validated ledger instances has a sequence number, which is one greater than the sequence number of the preceding instance. Each ledger also has an identifying hash value, which is uniquely determined from its contents. There may be many different versions of in-progress ledgers, which have the same sequence number, but different hash values. Only one version can ever be validated.
```
_Nitpick_
Perhaps: "There may be many different versions of in-progress ledgers, which have the same sequence number but different hash values. Only one version of the ledger can ever be validated."

mDuo13 replied:
Agreed. Also fixed the typo "Each validated ledger instances has..."

ryangyoung commented on content/reliable_tx.md:
```diff
@@ -28,33 +28,37 @@ Applications face additional challenges, in the event of power or network loss,
 Use the `LastLedgerSequence` parameter to prevent undesirable cases where a transaction is not promptly validated yet could become viable at some point in the future. Gateways and other back-end applications should specify the `LastLedgerSequence` parameter on every transaction. Automated processes should use a value of 4 greater than the sequence number of the last validated ledger[1] to ensure  ...
 
-Applications using rippled APIs should explicitly specify a `LastLedgerSequence` when submitting transactions.  When using ripple-lib or Ripple-REST, a `LastLedgerSequence` is automatically included.  Applications using those APIs are recommended to use the `LastLedgerSequence` calculated by the API.
+Applications using rippled APIs should explicitly specify a `LastLedgerSequence` when submitting transactions. RippleAPI uses the `maxLedgerVersion` field of [Transaction Instructions](rippleapi.html#transaction-instructions) to specify the `LastLedgerSequence`. RippleAPI automatically provides an appropriate value by default. You can specify `maxLedgerVersion` as `null` to intentionally omit `La ...
```
The maxLedgerVersion behavior is interesting. Should there be any warning about possible undesired consequences of submitting a transaction with no expiration? (Just asking; I don't know if there are any or what they might be.)

mDuo13 replied:
The consequences are as laid out in the opening paragraphs of the article -- 

> Applications which fail to implement best practices are at risk of the following errors:
> 1. Submitting transactions which are inadvertently never executed.
> 2. Mistaking provisional transaction results for their final, immutable results.
> 3. Failing to find authoritative results of transactions previously applied to the ledger.

I think that's sufficient.

ryangyoung replied:
LGTM.

ryangyoung commented on content/reliable_tx.md:
```diff
@@ -121,31 +125,31 @@ For each persisted transaction without validated result:
+    * Final results are an immutable part of the ledger history.
 
-An application's means of performing these actions depends on the ripple API the application uses.  These interfaces may be any of:
+An application's means of performing these actions depends on the API the application uses.  These interfaces may be any of:
```
_Nitpick_
Maybe something like "An application may use any of the following interfaces:"

ryangyoung commented on content/reliable_tx.md:
```diff
@@ -121,31 +125,31 @@ For each persisted transaction without validated result:
-3. [ripple-lib](https://github.com/ripple/ripple-lib/)
+1. [`rippled`'s internal APIs](rippled-apis.html)
+2. [RippleAPI](rippleapi.html)
+3. Any number of other software APIs layered on top of `rippled`
```
Without specifying, point 3 sounds kind of "hand-wavy". Are there any resources, an example, that we can point users to?

mDuo13 replied:
Point 3 is meant to include things like Ripple-REST, Ripple Connect, the Admin Console/API, and whatever else we make. I don't think we want to directly call any of these out here, thus the "hand-wavyness." I'm not really sure how to fix that.

ryangyoung replied:
That's what I suspected. That's fine for now.

ryangyoung commented on gateway_guide.html:
```diff
@@ -409,267 +403,295 @@ <h2 id="generating-source-and-destination-tags">Generating Source and Destinatio
 <li>Keep a record of the most-recently-processed transaction and ledger. That way, if you temporarily lose connectivity, you know how far to go back.</li>
 <li>Check the result code of every incoming payment. Some payments go into the ledger to charge an anti-spam fee, even though they failed. Only transactions with the result code <code>tesSUCCESS</code> can change non-XRP balances. Only transactions from a validated ledger are final.</li>
-<li><a href="https://ripple.com/files/GB-2014-06.pdf" title="Partial Payment Flag Gateway Bulletin">Look out for Partial Payments</a>. If an incoming transaction has a <code>destination_balance_changes</code> field (Ripple-REST) or a <code>meta.delivered_amount</code> field (WebSocket/JSON-RPC), then use that to see how much money <em>actually</em> got delivered to the destination account. Paymen ...
+<li><a href="https://ripple.com/files/GB-2014-06.pdf" title="Partial Payment Flag Gateway Bulletin">Look out for Partial Payments</a>. Payments with the partial-payment flag enabled can be considered "successful" if any non-zero amount is delivered, even miniscule amounts.<ul>
```
KC link for the gateway bulletin: https://ripple.com/knowledge_center/partial-payment-flag/

mDuo13 replied:
Yeah, this will be re-generated from the .md

ryangyoung replied:
PBKAC: didn't realize this was the HTML version of the same content.

ryangyoung commented on content/rippled.md:
```diff
@@ -8,7 +8,7 @@ The core peer-to-peer server that operates the Ripple Network is called `rippled
 
 # WebSocket and JSON-RPC APIs #
-If you want to communicate directly with the `rippled` server, you can use either the WebSocket API or the JSON-RPC API. Both APIs use the same list of commands, with almost entirely the same parameters in each command. Whereas the [Ripple-REST API](ripple-rest.html) provides a simplified interface on top of the WebSocket API for easier integration, these APIs provide the full power of Ripple but ...
+If you want to communicate directly with the `rippled` server, you can use either the WebSocket API or the JSON-RPC API. Both APIs use the same list of commands, with almost entirely the same parameters in each command.
```
You don't want to reference RippleAPI here?

alandotcom replied:
I agree, maybe update references for Ripple-Rest APIs to Ripple-API

mDuo13 replied:
Added.

alandotcom commented:
A few ripple-rest mentions still left:

```
content/rippled.md
7220:<!--Note: keep the above table up-to-date with the Get Server Status method in the Ripple-REST documentation -->

js/toc.js
16:        "title": "Ripple-REST API",
17:        "slug": "ripple-rest",

rippled-apis.html
4518:<!--Note: keep the above table up-to-date with the Get Server Status method in the Ripple-REST documentation -->
```

alandotcom commented:
There's also a few files left that should be deleted:

./js/apitool-rest.js
./tool/ripple-rest-markdown-skeleton.md
./tool/template-rest-api-tool.html

mDuo13 commented:
actually those last 3 files are not for Ripple-REST specifically but any REST API such as the Data API v2. I'll rename the skeleton to make that clear.

mDuo13 committed:
remove ripplerest - edits from reviews

mDuo13 commented:
to address @lumberj's other concerns:
1. Removed `js/toc.js` which was an unused file anyway
2. removed the comment about keeping Ripple-REST's server_info response synced with rippled's. That change flows from the .md to the .html version too.
3. Renamed `tool/ripple-rest-markdown-skeleton.md` to `rest-api-markdown-skeleton.md` since it's still useful for other RESTful APIs.
4. `tool/template-rest-api-tool.html` is still used for other RESTful APIs including Data API v2.
5. `js/apitool-rest.js` is still used for other RESTful APIs including Data API v2.

The PR should not be updated with all said changes.
